Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAssetHandler.cpp

```diff
@@ -36,14 +36,155 @@ namespace AZ
+        {
+            StreamingImageAsset* assetData = streamingImageAsset.GetAs<StreamingImageAsset>();
+            for (auto& mipChainAsset : assetData->m_mipChains)
+            {
+                // Need to skip tail mipchain which is in the list and doesn't have an asset id.
+                if (mipChainAsset.m_asset.GetId().IsValid())
+                {
+                    if (connect)
+                    {
+                        Data::AssetBus::MultiHandler::BusConnect(mipChainAsset.m_asset.GetId());
+                    }
+                    else
+                    {
+                        Data::AssetBus::MultiHandler::BusDisconnect(mipChainAsset.m_asset.GetId());
+                    }
+                }
+            }
+        }
+
+        void StreamingImageAssetHandler::HandleMipChainAssetLoad(Data::Asset<Data::AssetData> asset, bool isLoadSuccess)
+        {
+            AZStd::scoped_lock<AZStd::mutex> lock(m_accessPendingAssetsMutex);
+            auto itr = m_pendingReloadImageAsset.find(asset.GetId().m_guid);
+            if (itr == m_pendingReloadImageAsset.end())
+            {
+                return;
+            }
+
+            if (!isLoadSuccess)
+            {
+                HandleMipChainAssetBuses(itr->second.m_imageAsset, false);
+                Data::AssetManagerBus::Broadcast(&Data::AssetManagerBus::Events::OnAssetReloadError, itr->second.m_imageAsset);
+                m_pendingReloadImageAsset.erase(itr);
+                return;
+            }
+
+            uint32_t loadedMipChainAssets = 0;
+            for (u32& subId : itr->second.m_mipChainAssetSubIds)
+            {
+                if (subId == asset.GetId().m_subId)
+                {
+                    StreamingImageAsset* assetData = itr->second.m_imageAsset.GetAs<StreamingImageAsset>();
+
+                    // Assign the reloaded/loaded asset to StreamingImageAsset
+                    for (auto& mipChainAsset : assetData->m_mipChains)
+                    {
+                        if (mipChainAsset.m_asset.GetId() == asset.GetId())
+                        {
+                            mipChainAsset.m_asset = asset;
+                            // set the subId to 0 after it's loaded
+                            subId = 0;
+                            break;
+                        }
+                    }
+
+                }
+                if (subId == 0)
+                {
+                    loadedMipChainAssets++;
+                }
+            }
+
+            if (loadedMipChainAssets == itr->second.m_mipChainAssetSubIds.size())
+            {
+                HandleMipChainAssetBuses(itr->second.m_imageAsset, false);
+                Data::AssetManagerBus::Broadcast(&Data::AssetManagerBus::Events::OnAssetReloaded, itr->second.m_imageAsset);
+                m_pendingReloadImageAsset.erase(itr);
```

Comment:
Same here, might want to try and avoid holding the mutex across an arbitrary event broadcast.